src/System.Private.ServiceModel/tests/Common/Infrastructure/src/BridgeClient.cs

```diff
@@ -190,28 +192,64 @@ private static string CreateConfigRequestContentAsJson()
             return sb.ToString();
         }
 
-        private static string MakeResourcePutRequest(string resourceName)
+        private static string MakeEndpointResourcePutRequest(string resourceName)
+        {
+            var responseContent = MakeResourcePutRequest(resourceName, null);
+
+            string uri = null;
+            if (!responseContent.TryGetValue(EndpointResourceResponseUriKeyName, out uri) || String.IsNullOrWhiteSpace(uri))
+            {
+                StringBuilder sb = new StringBuilder();
+                foreach (var pair in responseContent)
+                {
```

Comment:
Nit: consider a helper that does this StringBuilder over a Dictionary<string, string>.  Not critical.